be/src/exprs/aggregate_functions.cpp

```diff
@@ -305,6 +250,59 @@ DoubleVal AggregateFunctions::percentile_approx_finalize(FunctionContext* ctx, c
     return DoubleVal(result);
 }
 
+struct AvgState {
+    double sum;
+    int64_t count;
+};
+
+struct DecimalAvgState {
+    DecimalVal sum;
+    int64_t count;
+};
+
+struct DecimalV2AvgState {
+    DecimalV2Val sum;
+    int64_t count = 0;
+};
+
+void AggregateFunctions::avg_init(FunctionContext* ctx, StringVal* dst) {
+    dst->is_null = false;
+    dst->len = sizeof(AvgState);
+    dst->ptr = ctx->allocate(dst->len);
+    memset(dst->ptr, 0, sizeof(AvgState));
```

Comment:
better to give default value to AvgState, and call placement new to initialize it.